### PR TITLE
*: fix etcd and kenc container command

### PR DESCRIPTION
**Pull request by hongchaodeng**
1. get rid of `sh`
2. fix `flock` command

### CODE REVIEW

**Comment by hongchaodeng** [comment]
@xiang90 @aaronlevy

File: pkg/util/etcdutil/start_etcd.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ spec:
   containers:
   - command:
     - /bin/sh
-    - -c
+    - -ec
```
Comment (aaronlevy) [inline pkg/util/etcdutil/start_etcd.go]: Why do we need to wrap this in a shell here? I think we could execute this directly. FWIW when you use $(FOO) the kubelet will template in those values (they are not being used as environment variables - note the `"()"` vs `"{}"`

File: pkg/asset/internal/templates.go [hunk 2]
```diff
@@ -233,8 +233,8 @@ spec:
           readOnly: false
         command:
         - "/bin/sh"
-        - "-c"
-        - "/usr/bin/flock --exclusive --timeout=30 /var/lock/kenc.lock; kenc -r -m iptables && kenc -m iptables"
+        - "-ec"
```
Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: Aso doesn't seem like this needs to be wrapped in a shell. Can't we just execute flock directly?

File: pkg/util/etcdutil/start_etcd.go (hunk 1 above)
Comment (hongchaodeng) [inline pkg/util/etcdutil/start_etcd.go]: Just my pick. I usually wrap everything in `sh -ec`. Then the whole command could be run in a single string.

I can change if you want.

Comment (aaronlevy) [inline pkg/util/etcdutil/start_etcd.go]: Yeah, if we don't need separate shell - seems simpler to just execute the binary directly.

Comment (hongchaodeng) [inline pkg/util/etcdutil/start_etcd.go]: Also the `$(..)` templating still works under `sh -ec`

**Comment by hongchaodeng** [comment]
Updated

**Comment by pbx0** [comment]
@hongchaodeng The change to the boot-etcd.yaml is causing things to fail. I'm not exactly sure why but the docker logs for that container says:
```
flag provided but not defined: -name boot-etcd
usage: etcd [flags]
       start an etcd server

       etcd --version
       show the version of etcd

       etcd -h | --help
       show the help information about etcd

       etcd --config-file
       path to the server configuration file
```

**Comment by hongchaodeng** [comment]
fixed

**Comment by pbx0** [comment]
rktbot run tests

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by pbx0** [comment]
@hongchaodeng The CI tests look good to me. The self-hosted scale test is struggling these days, but the other tests all pass including the self-hosted smoke.

**Comment by hongchaodeng** [comment]
@aaronlevy PTAL

**Comment by aaronlevy** [comment]
lgtm

**Pull request merged**